src/Middleware/OutputCaching/src/CacheVaryByRules.cs

```diff
@@ -20,10 +20,15 @@ public sealed class CacheVaryByRules
     /// </summary>
     public IDictionary<string, string> VaryByCustom => _varyByCustom ??= new();
 
+    /// <summary>
+    /// Gets or sets the list of route values to vary by.
+    /// </summary>
+    public StringValues RouteValues { get; set; }
```

Comment:
Why was `StringValues` chosen over `IList<string>`? Seems like an odd choice.

Reply:
For consistency I kept the same as in ResponseCaching because they probably had reasons. There was no comment about it during api review either. I have no opinion, not sure about perf impact. It does make the code easier though since we don't have to check for nulls before accessing it, whereas with a list we'd need to lazy initialize it when accessed. And check an internal field to see if a value is set.